modules/core/CameraStereoscopic.cpp

```diff
@@ -0,0 +1,28 @@
+#include "CameraStereoscopic.h"
+#include "Ray.h"
+#include "macroses.h"
+#include "CameraEnvironmentTarget.h"
+
+namespace rt
+{
+void CCameraStereoscopic::InitRay(Ray& ray, int x, int y, const Vec2f& sample)
+    {
+        Size resolution = getResolution();
+        const Vec3f& direction = normalize(m_dir - m_pos);
+        const Vec3f& translation_vector = normalize(direction.cross(m_up));
+        
+        if(m_eye == 1) {
+            const Vec3f& left_position = m_pos - .5f * m_IPD * translation_vector;
+            CCameraEnvironmentTarget *Left = new CCameraEnvironmentTarget(resolution, left_position, m_dir, m_up);
```

Comment:
Here is the memory leakage.
You create a new camera for every single ray. If you need to do so, it is better to create environmental camera only once _e.g._ in constructor.